pkg/gds/tokens/mock.go

```diff
@@ -25,3 +30,40 @@ func MockTokenManager() (tm *TokenManager, err error) {
 	tm.keys[tm.currentKeyID] = &key.PublicKey
 	return tm, nil
 }
+
+// mockValidate is a mock for Google's token validation which enables authentication
+// testing by internally parsing the given token and injecting the claims into the
+// returned payload. This will only validate badly constructed tokens and tokens where
+// the supplied audience does not match the audience used to sign the token.
```

Comment:
You mention that this mock will only allow us to catch bad tokens and mismatched audiences; are there other edge cases we should be thinking about moving forward?

Reply:
Good question, we could also verify that the expiration time and issued time make sense which isn't that complicated of a check. Currently we're passing basically all of the token validation off to Google so from a testing perspective there's only really one validation error path, so I didn't find it absolutely necessary to include those checks in the mock for the scope of what this PR is testing.